pentasharp/Controllers/TaxiBookingController.cs

```diff
@@ -0,0 +1,142 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using pentasharp.Data;
+using pentasharp.Models.DTOs;
+using pentasharp.Models.Entities;
+using pentasharp.Models.Enums;
+using pentasharp.Models.Utilities;
+using pentasharp.ViewModel.Taxi;
+using pentasharp.ViewModel.TaxiBooking;
+using pentasharp.ViewModel.TaxiModels;
+using WebApplication1.Filters;
+
+namespace WebApplication1.Controllers
+{
+    [Route("api/TaxiBooking")]
+    public class TaxiBookingController : Controller
+    {
+        private readonly AppDbContext _context;
+        private readonly IMapper _mapper;
+
+        public TaxiBookingController(AppDbContext context, IMapper mapper)
+        {
+            _context = context;
+            _mapper = mapper;
+        }
+
+        [AllowAnonymous]
+        [HttpPost("SearchAvailableTaxis")]
+        public IActionResult GetCompanies()
```

Comment:
move the business logic you are doing in the controllers to a service, create its interface, register the service in program.cs and inject it propely in the controller. Same as per the comment i made in Oltas PR.